src/FormModel.php

```diff
@@ -345,7 +356,13 @@ private function readProperty(string $attribute)
 
     private function writeProperty(string $attribute, $value): void
     {
-        if ($this->isPublicAttribute($attribute)) {
+        $setter = 'set' . $this->inflector->camelize($attribute);
+        if (is_callable([$this, $setter])) {
+            $method = new ReflectionMethod(static::class, $setter);
+            if ($method->getNumberOfRequiredParameters() === 1) {
+                $this->$setter($value);
+            }
+        } else if ($this->isPublicAttribute($attribute)) {
```

Comment:
```suggestion
        } elseif ($this->isPublicAttribute($attribute)) {
```

Reply:
The resolution will depend on the result of [discussion](https://github.com/yiisoft/form/pull/38#discussion_r470771047) above.